app/scripts/lib/segment.js

```diff
@@ -1,9 +1,6 @@
 import Analytics from 'analytics-node';
 import { SECOND } from '../../../shared/constants/time';
 
-const isDevEnvironment = Boolean(
-  process.env.METAMASK_DEBUG && !process.env.IN_TEST,
-);
```

Comment:
@Gudahtt i could be mistaken, but we have a segment write key set in CIRCLE CI right? Is there any risk if we don't do the IN_TEST check that this hits segment?

Reply:
Ah, great point. We do indeed have this set in CircleCI, so that the metamaskbot builds send events to the dev segment instance.

Removing this check would send the test build events to the dev segment instance as well, which seems bad.

Reply:
Could be fixed by changing the condition below from `SEGMENT_WRITE_KEY` to `SEGMENT_WRITE_KEY && !process.env.IN_TEST`

Reply:
Sorry, trying to wrap my head around this more. Does this mean we want to send events for metamaskbot builds in CircleCL and not our test build? In which case, adding back
```
const isDevEnvironment = Boolean(
  process.env.METAMASK_DEBUG && !process.env.IN_TEST,
);
```
and checking `SEGMENT_WRITE_KEY && !isDevEnvironment` would be what we want? 

@brad-decker @Gudahtt

Reply:
I made the update in case the above is what we want. I can change it to `SEGMENT_WRITE_KEY && !process.env.IN_TEST` [as mentioned](https://github.com/MetaMask/metamask-extension/pull/14407#discussion_r846330536) if I understand what the env variables are doing some more

Reply:
Thank you very much for explaining this, and sorry for the mixup here in the last couple of comments. Honestly, I reread this a few times to fully understand it, and I think I have it now. 

To confirm, we want to check `!IN_TEST` because this is related to the test builds, `Test dev build` and `Test build` that you are mentioning. 

From what I understand, another way to look at the other builds is:
- Dev
- Test
  - Dev
  - CI 
- Environment
  - Prod
  - Beta 
  - Prod - Flask
  - etc.
  
I've updated the code to implement your [original suggestion](https://github.com/MetaMask/metamask-extension/pull/14407#discussion_r846330536)

Reply:
😅  oops. Okay, continuing this convo in [the other thread](https://github.com/MetaMask/metamask-extension/pull/14407#discussion_r850498894)